crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -150,6 +193,23 @@ pub(crate) fn highlight(
             };
             let token = sema.descend_into_macros(token.clone());
             let parent = token.parent();
+
+            // Check if macro takes a format string and remember the string
+            if let Some(name) = current_macro_call
+                .as_ref()
+                .and_then(|m| m.path())
+                .and_then(|p| p.segment())
+                .and_then(|s| s.name_ref())
+            {
+                if ["format", "println", "print", "eprint", "eprintln", "write", "writeln"]
```

Comment:
Yeah, the fancier option @matklad mentions is to check if it's expanded to builtin macro [`std::format_args`](https://doc.rust-lang.org/std/macro.format_args.html)

Reply:
Thanks! Searching for `std::format_args` is a way better solution and would for example also work with macros from `log`. However, I was not able to expand the macros. I tried the `Semantics::expand` and `Semantics::resolve_macro_call` but they always return `None`.

Reply:
I didn't test it, but I think the following logic would work (probably) :

Since ` token` is descended (by `descend_into_macros`) and if I remember correctly, we don't descend a token into builtin-macro such that you could just check whether if `token` ancestor is `format_arg` directly. 

But I might be wrong here.

Reply:
I have tried that, but it does not work, the direct ancestor is the original/non-expanded macro. This is probably because `println!` is a std macro as well?

Reply:
`println` is from std, but it's a normal macro. `format_args` is a compiler built-in. One gotcha might be that this won't work in unit tests unless you define the `println` macro. Did you try it in a unit test, or on real code?

Reply:
Yes, I only tried in a unit test. 
I have updated to PR to include more std macros (and their correct argument positions). I agree that the fancier solution is probably better but I was not able to get it right 😉

Reply:
Here's how you would get the fancier solution working in a unit test: Copy the definitions of the relevant macros to the test code, e.g.
```rust
macro_rules! println {
    ($($arg:tt)*) => ({
        $crate::io::_print($crate::format_args_nl!($($arg)*));
    })
}
#[rustc_builtin_macro]
macro_rules! format_args_nl {
    ($fmt:expr) => {{ /* compiler built-in */ }};
    ($fmt:expr, $($args:tt)*) => {{ /* compiler built-in */ }};
}
```
(copying the definition of `_print` isn't necessary, since we just need the macros to expand.
 Copying the `format_args_nl` definition is probably also not really necessary if you just test for the name, but slightly more realistic.)
Unit tests don't have access to std (or core), just the stuff built-in to the compiler, mostly because it would slow down unit tests a lot if every one of them would have to parse all of std.

Reply:
Thanks for the clarification! I've updated the PR to use the `format_args` detection method. However, this is not perfect, e.g., `assert!` is itself a compiler builtin macro and is thus not detected.